src/elements/common/with-router-and-ref/withRouterAndRef.js

```diff
@@ -0,0 +1,13 @@
+// @flow
+import * as React from 'react';
+import { withRouter } from 'react-router-dom';
+
+export default function withRouterAndRef(Wrapped: React.ComponentType<any>) {
+    const WithRouter = withRouter(({ forwardRef, ...otherProps }) => <Wrapped ref={forwardRef} {...otherProps} />);
+    const WithRouterAndRef = React.forwardRef<Object, React.Ref<any>>((props, ref) => (
+        <WithRouter {...props} forwardRef={ref} />
+    ));
```

Comment:
Can we combine these two?

Reply:
Hmm I'm not sure, what did you have in mind?

Reply:
Never mind, I just ended up moving `WithRouter` into `WithRouterAndRef` body, which is not elegant.